ui/src/widgets/ui-text-input/UITextInput.vue

```diff
@@ -110,7 +109,36 @@ export default {
             }
         }
     },
+    created () {
+        // can't do this in setup as we are using custom onInput function that needs access to 'this'
+        useDataTracker(this.id, this.onInput, this.onLoad, null)
+
+        // let Node-RED know that this widget has loaded
+        this.$socket.emit('widget-load', this.id)
+    },
     methods: {
+        onInput (msg) {
+            // update our vuex store with the value retrieved from Node-RED
+            this.$store.commit('data/bind', {
+                widgetId: this.id,
+                msg
+            })
```

Comment:
I'd argue here that actually, we only want to store this _if_ a `payload` is present? This encourages the argument for us using `datastore` as a bigger picture state, because here, we'd only get the latest `msg.enabled` and not the underlying `payload`

Reply:
It was another copy (honestly don't know what is being stored in Vuex and for what is used), but not all the widgets have the `onInput`.

Will change to only store, if `msg.payload` is present in the message.

Reply:
Just did some tests, and honestly don't quite understand to point of this.

Tried to pass code the "this.$store.commit" for only if `payload` exists, but the behaviour is the same, if a message arrives to the widget on NodeRed that message is "saved" and replayed later on "widget-load", and the `payload` may not appear if it wasn't in that message (in case of page refresh for example). The message seems to be saved here:
https://github.com/FlowFuse/node-red-dashboard/blob/f0a9cad51cf16655419b4d32ee4bf3f8b56d0bfa/nodes/widgets/ui_text_input.js#L20
And it is replayed, the saving (commit) to the `vuex` store seems to have no effect